code/game/objects/items/explosives/grenades/marines.dm

```diff
@@ -137,6 +157,16 @@
 	var/turf/T = get_turf(src)
 	T.ignite(25, 25)
 
+/obj/item/explosive/grenade/sticky/trailblazer/afterattack(atom/target, mob/user)
+	. = ..()
+	if(target != user)
+		return
+	if(self_sticky == FALSE)
+		return
```

Comment:
why the copy paste checks?
make the parent return true if it self sticks, then just make an early return on !.

Reply:
> why the copy paste checks? make the parent return true if it self sticks, then just make an early return on !.

pushed some more stuff, this is the best I can do with my knowledge, I'd need walked through how to make it any cleaner, messed around with it for awhile and just ended up making it worse for awhile